src/main/java/messaging/MessagePresenter.java

```diff
@@ -0,0 +1,37 @@
+package messaging;
+import javax.swing.*;
+
+import ui.View;
+
+public class MessagePresenter implements MessageOutputBoundary {
+    private View view;
+
+    @Override
+    public void prepareRoomView(String messageHistory) {
+        view.setMessageHistory(messageHistory);
+        view.prepareGUI();
+    }
+
+    @Override
+    public void prepareMessageErrorView() {
+        JOptionPane.showMessageDialog(null, "Error sending message, cannot send empty message!", "Message Error", JOptionPane.ERROR_MESSAGE);
```

Comment:
We disallow empty messages from the view so maybe here we should have a different error message?

Reply:
Not entirely sure what you mean, should I just change the error message?